web_src/css/repository.css

```diff
@@ -1846,6 +1846,25 @@
   padding-bottom: 5px;
 }
 
+.diff-file-box {
+  border: 1px solid transparent;
+  border-radius: 0.285rem; /* Just like ui.top.attached.header */
+}
+
+/* TODO: try making these rules global to fix elements breaking out of their parent box */
+.diff-file-box .ui.attached.header,
+.diff-file-box .ui.attached.table {
+  margin: 0; /* remove fomantic negative margins */
+  width: initial; /* remove fomantic over 100% width */
+  max-width: initial; /* remove fomantic over 100% width */
+}
+
+.diff-file-box:target {
+  border-color: var(--color-primary) !important;
+  border-radius: var(--border-radius) !important;
+  box-shadow: 0 0 0 3px var(--color-primary-alpha-30) !important;
```

Comment:
Why not `--color-accent` anymore?
That's the semantically correct color here…

Reply:
> Use the same border/box-shadow combo used on comment :target also for file :target.

Or better: Why not convert the other color to `--color-accent` as well?

Reply:
The style is the exact same as from comment highlight. The box shadow requires an alpha channel color, but without CSS [`color-mix`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix) I'm afraid we can not automatically derive this alpha-channel color. So I suggest leave it like that.

Reply:
BTW to me the distinction between `--color-accent` and `--color-primary` is not clearly defined. For all intends and puposes, they should be the same color imho.

Reply:
I think I might even merge the rules for comment and file highlight into one so it's clear they should be the same colors.

Reply:
Well, for me, the difference is the following:
- `color-primary`: UI elements that are supposed to be interacted (`Ok`/`Save` buttons, …)
- `color-accent`: UI elements that should receive a highlighting, but the highlighting is purely for stylistic purposes, and it is not said, if you can interact with this element…

Reply:
As per [MDN](https://developer.mozilla.org/en-US/docs/Glossary/Accent):

> An accent is a typically bright color that contrasts with the more utilitarian background and foreground colors within a color scheme.

> On the web, an accent is sometimes used in [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) elements for the active portion of the control, for instance the background of a checked [checkbox](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox).

I think the use case exactly matches that of `--primary-color`, so I see no reason to make them different. Accent has nothing to do with highlight imho.

Reply:
Merged the rules in https://github.com/go-gitea/gitea/pull/23553/commits/6f295f13f9ee71614821a0a17fd212521a4bcaad. I think we should leave those as `--primary-color`. When CSS `color-mix` is available, we could re-evaluate deriving alpha-channel colors off it and maybe apply it to the rule then, but currently this needs extra CSS vars which is problematic for themes.